### ISSUE TITLE
Create ElastiCache Resource

### ISSUE THREAD

**Issue by christopherhein** [awslabs/aws-service-operator#37]
This resource will allow a developer to create an ElastiCache instance using
the CRD model, it should be generic enough to support both memcached and redis.

```yaml
apiVersion: operator.aws/v1alpha1
kind: ElastiCache
metadata:
  name: elasticache
spec:
  autoMinorVersionUpgrade: true
  azMode: # (single-az|cross-az) only memcached
  nodeType: cache.m3.medium
  engine: # (memcached|redis)
  version: 4.0.10
  cacheNodes: 3
  tags:
  - name: Usage
    value: caching
```

**Comment by jessebye** [comment]
This issue should be closed since this is now implemented!

**Comment by christopherhein** [comment]
Thanks @jessebye, you’re absolutely correct!

**Comment by dathan** [comment]
It would be nice if the elastic cache permissions where in the repo as well? For instance, the default role in the example does not give ec2 resources nor does it document you need to create a service linked role.

**Comment by christopherhein** [comment]
@dathan great suggestion. I'd like to have this solved using #59 and #58 but for now I haven't been thinking about it mutating the ec2 permissions. Given that would grant any node that uses the role the same permissions. We have a longer term goal to solve IAM to Pod documented https://github.com/aws/containers-roadmap/issues/23 and what I'm hoping is that the operator can automate the creation and management of the policies while IAM to Pod will solve the handing credentials to the workloads.
